library/cohesity_oracle.py

```diff
@@ -0,0 +1,415 @@
+
+        # Payload template to register a physical source.
+        payload = {
+            "entity": {
+                "type": 6,
+                "physicalEntity": {
+                    "name": "",
+                    "type": 1,
+                    "hostType": 1
+                }
+            },
+            "entityInfo": {
+                "endpoint": "",
+                "type": 6,
+                "hostType": 1
+            },
+            "sourceSideDedupEnabled": True,
+            "throttlingPolicy": {
+                "isThrottlingEnabled": False
+            },
+        }
+        payload["forceRegister"] = force_register
+        payload["entity"]["physicalEntity"]["name"] = endpoint
+        payload["entityInfo"]["endpoint"] = endpoint
+
+        data = json.dumps(payload)
+        response = open_url(url=uri, data=data, headers=headers,
+                            validate_certs=validate_certs, timeout=REQUEST_TIMEOUT)
+
+        response = json.loads(response.read())
+
+        # => This switcher will allow us to return a standardized output
+        # => for all Protection Sources.
+        # response = dict(ProtectionSource=response[
+        #                     'physicalProtectionSource'])
+        return dict(response)
+    except urllib_error.URLError as e:
+        # => Capture and report any error messages.
+        raise__cohesity_exception__handler(e.read(), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+# => Unregister an existing Cohesity Protection Source.
+def unregister_source(module, self):
+    server = module.params.get('cluster')
+    validate_certs = module.params.get('validate_certs')
+    token = self['token']
+    try:
+        uri = "https://" + server + \
+            "/irisservices/api/v1/public/protectionSources/" + str(self['id'])
+        headers = {"Accept": "application/json",
+                   "Authorization": "Bearer " + token}
+
+        response = open_url(url=uri, method='DELETE', headers=headers,
+                            validate_certs=validate_certs, timeout=REQUEST_TIMEOUT)
+
+        return response
+    except urllib_error.URLError as e:
+        # => Capture and report any error messages.
+        raise__cohesity_exception__handler(e.read(), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def main():
+    # => Load the default arguments including those specific to the Cohesity Agent.
+    argument_spec = cohesity_common_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            endpoint=dict(type='str', required=True),
+            # => Currently, the only supported environments types are list in the choices
+            # => For future enhancements, the below list should be consulted.
+            # => 'SQL', 'View', 'Puppeteer', 'Pure', 'Netapp', 'HyperV', 'Acropolis', 'Azure'
```

Comment:
Is this accurate ? @anvesh-cohesity

Reply:
can remove that, not sure why these comments are added here

Reply:
Removed.